torch/functional.py

```diff
@@ -989,16 +1009,38 @@ def tensordot(a, b, dims=2, out=None):
     """
     if has_torch_function_variadic(a, b):
         return handle_torch_function(tensordot, (a, b), a, b, dims=dims)
-    if isinstance(dims, (list, tuple)) or \
-       (isinstance(dims, torch.Tensor) and dims.numel() > 1):
+
+    dims_a: List[int] = []
+    dims_b: List[int] = []
+
+    if isinstance(dims, tuple):
         dims_a, dims_b = dims
-    else:
-        if isinstance(dims, torch.Tensor):
-            dims = dims.item()
+
+    if isinstance(dims, list):
```

Comment:
Nit you can fold this if with the tuple one.
if isinstance(dims, (list, tuple))

Reply:
Done.